aten/src/ATen/native/cuda/UpSampleNearest2d.cu

```diff
@@ -145,26 +149,42 @@ static void upsample_nearest2d_out_cuda_template(
       input_height > 0 && input_width > 0 && output_height > 0 &&
       output_width > 0);
 
-  output.resize_({input.size(0), input.size(1), output_height, output_width});
+  Tensor input = input_.contiguous();
+  output.resize_({nbatch, channels, output_height, output_width});
   output.zero_();
```

Comment:
Do you need .zero_() here? Output is overwritten, not accumulated, I think

Reply:
Good catch!
it was needed for previous implementation with atomicAdd. Hence it's now obsolete. I'll remove it.